src/main/scala/io/iohk/ethereum/blockchain/sync/FastSyncNodesRequestHandler.scala

```diff
@@ -76,49 +79,56 @@ class FastSyncNodesRequestHandler(
   }
 
   private def handleMptNode(mptNode: MptNode): Seq[HashType] = mptNode match {
```

Comment:
This function is almost the same as `handleContractMptNode`, especially `ExtensionNode` and `BranchNode` cases, maybe w can unify it somehow.
I was thinking something along these lines for extension node in both of them:
``` 
private def handleExtensionNode(n: ExtensionNode)(f: ByteString => Seq[HashType]): Seq[HashType] = {
    saveNodeFn(ByteString(n.hash), n.toBytes)
    n.next.fold(f, _ => Nil)
  }
```